numpy/linalg/linalg.py

```diff
@@ -1010,8 +991,6 @@ def eigvalsh(a, UPLO='L'):
     _assertRankAtLeast2(a)
     _assertNdSquareness(a)
     t, result_t = _commonType(a)
-    if _isEmpty2d(a):
```

Comment:
There was a bug here  for complex value, since it missed the cast to real (I think in one other function as well).